packages/invoices/src/InboundShipment/DetailView/api.ts

```diff
@@ -201,31 +127,31 @@ export const getInboundShipmentDetailViewApi = (
   onRead: async (id: string): Promise<Invoice> => {
     const result = await api.invoice({ id, storeId });
 
-    const invoice = invoiceGuard(result);
-    const lineNodes = linesGuard(invoice.lines);
-    const lines: InvoiceLine[] = lineNodes.map(line => {
-      const stockLine = line.stockLine
-        ? stockLineGuard(line.stockLine)
-        : undefined;
-      const location = line.location ? locationGuard(line.location) : undefined;
-      const item = line.item ? itemGuard(line.item) : undefined;
+    const invoice = result.invoice;
+
+    if (invoice.__typename === 'InvoiceNode') {
```

Comment:
this is the `invoiceGuard` fn, just inlined as queries for specific entities - i.e. `query invoice` return a union of two members: `NodeError | InvoiceNode` for errors like `RecordNotFound`.

Seems fair enough! The fields however, all are returned as concrete types, so none of the other guards are needed